apps/browser/qml/pages/HistoryPage.qml

```diff
@@ -62,6 +62,24 @@ Page {
             }
         }
 
+        PullDownMenu {
+            flickable: view
+
+            MenuItem {
+                //% "Clear history"
+                text: qsTrId("sailfish_browser-me-clear-history")
+                onClicked: {
+                    Remorse.popupAction(
```

Comment:
Add "property var remorse" next to "property alias model" at the beginning.

Reply:
Will added other comments here as those are for lines outside of this changes...

1) To root level add: ```readonly property bool pendingRemorse: remorse && remorse.pending```

2) Change root level model to ```property QtObject model```

3) Change HistoryList model to ```model: pendingRemorse ? null : root.model```

4) To SearchField add: ``` enabled: !pendingRemorse && view.model && view.model.count > 0```

5) To ViewPlaceholde: ```enabled: view.pendingRemorse || !model.count```

6) For PullDownMenu: ```visible: model && model.count```

So idea is that when remorse popup is running we should treat the UI bits so that the action is already executed.